pkg/syndesis/action/attach.go

```diff
@@ -0,0 +1,161 @@
+package action
+
+import (
+	"github.com/operator-framework/operator-sdk/pkg/sdk"
+	"github.com/sirupsen/logrus"
+	"github.com/syndesisio/syndesis-operator/pkg/apis/syndesis/v1alpha1"
+	"github.com/syndesisio/syndesis-operator/pkg/syndesis/configuration"
+	"github.com/syndesisio/syndesis-operator/pkg/util"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// Attach a detached Syndesis installation to the Syndesis custom resource
+type Attach struct {}
+
+
+func (a *Attach) CanExecute(syndesis *v1alpha1.Syndesis) bool {
+	return syndesisInstallationStatusIs(syndesis, v1alpha1.SyndesisInstallationStatusAttaching)
+}
+
+func (a *Attach) Execute(syndesis *v1alpha1.Syndesis) error {
+
+	// Checking that there's only one installation to avoid stealing resources
+	if anotherInstallation, err := isAnotherActiveInstallationPresent(syndesis); err != nil {
+		return err
+	} else if anotherInstallation {
+		target := syndesis.DeepCopy()
+
+		target.Status.InstallationStatus = v1alpha1.SyndesisInstallationStatusNotInstalled
+		target.Status.Reason = v1alpha1.SyndesisStatusReasonDuplicate
+		target.Status.Description = "Cannot merge because another Syndesis resources is present in the same namespace"
+
+		logrus.Error("Cannot merge resource ", syndesis.Name, " because another Syndesis resources is present in the same namespace")
+		return sdk.Update(target)
+	}
+
+	logrus.Info("Attaching Syndesis installation to resource ", syndesis.Name)
+
+	for _, selector := range getAllManagerSelectors() {
+		for _, metaType := range getAllManagedResourceTypes() {
+
```

Comment:
wouldn't it be better to iterate over all the objects stored in the template ? It looks a bit dangerous and maintenance intense to hardcode the objects contained in the template here.

Up to now we only know about the Route in the template. BTW, what do you think about removing the Syndesis `Route` object from the template, and just create it as an extra object, either directly from the go-code or as an included yml ? That way we would avoid that dance with instantiating the template twice, fishing out the Route object etc. would be much directer. Even when we want to backport that to our manual installation, it would be easy to achieve and we could benefit from the automatically set route. I.e. we should handle the route on the same footstep as the service account as we need generated values from both (token, hostName) to instantiate the template.

Reply:
I didn't iterate over the template to avoid problems for things that for some reasons are removed from the template or change their name starting from some version. I may take "object types" from the template and leave room for adding other types if needed.. that would be less error prone..

About removing the Route from the template, that would simplify a lot the code as you've noticed.